make/buf-push-action/all.mk

```diff
@@ -0,0 +1,47 @@
+GO_BINS := $(GO_BINS) cmd/buf-push-action
+
+LICENSE_HEADER_LICENSE_TYPE := apache
+LICENSE_HEADER_COPYRIGHT_HOLDER := Buf Technologies, Inc.
+LICENSE_HEADER_YEAR_RANGE := 2020-2022
+LICENSE_HEADER_IGNORES := \/testdata
+FILE_IGNORES := $(FILE_IGNORES) .build/
+
+# Set a default value for OPEN_CMD so that _assert_var,OPEN_CMD doesn't fail in go.mk if the
+# OS is neither Darwin nor Linux.
+OPEN_CMD := open
+
+include make/go/bootstrap.mk
+include make/go/go.mk
+include make/go/license_header.mk
+include make/go/dep_minisign.mk
+
+bufgeneratedeps::
+
+.PHONY: bufgeneratecleango
+bufgeneratecleango:
+
+bufgenerateclean:: bufgeneratecleango
+
+.PHONY: bufgeneratego
+bufgeneratego:
+
+bufgeneratesteps:: bufgeneratego
+
+.PHONY: release
+release: $(MINISIGN)
+	DOCKER_IMAGE=golang:1.17.7-buster bash make/buf-push-action/scripts/release.bash
+
+.PHONY: directoryhash
+directoryhash:
+	@ bash make/buf-push-action/scripts/directoryhash.bash .
```

Comment:
It does hash the root of the repo, but `.git/` isn't present when this runs as an action. GitHub does the equivalent of downloading and extracting a tarball of the repository contents.

However after typing this, I realize that isn't documented behavior so it is subject to change. I should update `directoryhash.bash` to exclude `.git/` in case it ever shows up.